rs/linux_kernel_command_line/src/lib.rs

```diff
@@ -0,0 +1,354 @@
+                "rd.initrd=/bin/bash rd.escaped=\"this is a multiline argument\"".to_string(),
+            ),
+            (
+                "remove argument with quoted value in the middle of command line succeeds",
+                "rd.initrd=/bin/bash rd.debug=\"i am quoted value\" rd.escaped=\"this is a multiline argument\"",
+                "rd.debug",
+                "rd.initrd=/bin/bash rd.escaped=\"this is a multiline argument\"".to_string(),
+            ),
+            (
+                "remove argument with quoted value at the end of command line succeeds",
+                "rd.initrd=/bin/bash rd.escaped=\"this is a multiline argument\" rd.debug=\"i am quoted value\"",                "rd.debug",
+                "rd.initrd=/bin/bash rd.escaped=\"this is a multiline argument\"".to_string(),
+            ),
+            (
+                "argument with substring does not get removed at end of string",
+                "rd.initrd=/bin/bash rd.escaped=\"this is a multiline argument\" rd.debug=\"i am quoted value\"",                "rd.debu",
+                "rd.initrd=/bin/bash rd.escaped=\"this is a multiline argument\" rd.debug=\"i am quoted value\""
+                    .to_string(),
+            ),
+            (
+                "argument removal chomps extra spaces after removal",
+                "rd.initrd=/bin/bash  rd.escaped=\"this is a multiline argument\" rd.debug=\"i am quoted value\"",                "rd.debug",
+                "rd.initrd=/bin/bash rd.escaped=\"this is a multiline argument\""
+                    .to_string(),
+            ),
+        ];
+        for (name, input, argument_to_remove, expected) in table.iter() {
+            let mut cmdline = KernelCommandLine::try_from(*input).unwrap();
+            cmdline.remove_argument(*argument_to_remove);
+            let result: String = cmdline.into();
+            if result != *expected {
+                panic!(
+                    "During test {}:\ninput:    {:?}\nargument: {:?}\nexpected: {:?}\nactual:   {:?}",
+                    name, input, argument_to_remove, expected, result
+                );
+            }
+            println!("{:?}", result);
+        }
+    }
+
+    #[test]
+    fn test_unquoted_argument() {
+        let table = [
+            (
+                "misquoted argument at the beginning of command line succeeds",
+                "rd.debug=\"misquoted rd.initrd=/bin/bash rd.escaped=\"this is a multiline argument\""
+            ),
+            (
+                "misquoted argument in the middle of command line succeeds",
+                "rd.initrd=/bin/bash rd.debug=\"misquoted rd.escaped=\"this is a multiline argument\""
+            ),
+            (
+                "misquoted argument at the end of command line succeeds",
+                "rd.initrd=/bin/bash misquoted=\"true it's misquoted rd.escaped=\"this is a multiline argument\" rd.debug"
+            ),
+        ];
+        for (name, input) in table.iter() {
```

Comment:
nit: you can iterate directly without `.iter()`, 
`for (name, input) in table`

Reply:
No, that requires me to add references to the stuff within the for loop.  This is fine as is.

Reply:
FYI: 

```
Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
error[E0277]: the trait bound `KernelCommandLine: TryFrom<str>` is not satisfied
   --> rs/linux_kernel_command_line/src/lib.rs:257:31
    |
257 |             let mut cmdline = KernelCommandLine::try_from(*input).unwrap();
    |                               ^^^^^^^^^^^^^^^^^ the trait `From<str>` is not implemented for `KernelCommandLine`, which is required by `KernelCommandLine: TryFrom<str>`
    |
    = help: the trait `TryFrom<&str>` is implemented for `KernelCommandLine`
    = help: for that trait implementation, expected `&str`, found `str`
    = note: required for `str` to implement `Into<KernelCommandLine>`
    = note: required for `KernelCommandLine` to implement `TryFrom<str>`

error[E0277]: the size for values of type `str` cannot be known at compilation time
   --> rs/linux_kernel_command_line/src/lib.rs:257:59
    |
257 |             let mut cmdline = KernelCommandLine::try_from(*input).unwrap();
    |                               --------------------------- ^^^^^^ doesn't have a size known at compile-time
    |                               |
    |                               required by a bound introduced by this call
    |
    = help: the trait `Sized` is not implemented for `str`
note: required by a bound in `try_from`
   --> /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/core/src/convert/mod.rs:687:5

error[E0277]: the size for values of type `str` cannot be known at compilation time
   --> rs/linux_kernel_command_line/src/lib.rs:257:31
    |
257 |             let mut cmdline = KernelCommandLine::try_from(*input).unwrap();
    |                               ^^^^^^^^^^^^^^^^^ doesn't have a size known at compile-time
    |
    = help: the trait `Sized` is not implemented for `str`, which is required by `KernelCommandLine: TryFrom<str>`
    = help: the trait `TryFrom<&str>` is implemented for `KernelCommandLine`
    = help: for that trait implementation, expected `&str`, found `str`
    = note: required for `KernelCommandLine` to implement `TryFrom<str>`
```